packages/jest-cli/src/TestRunner.js

```diff
@@ -238,10 +247,22 @@ class TestRunner {
         aggregatedResults.success =
           aggregatedResults.numFailedTests === 0 &&
           aggregatedResults.numRuntimeErrorTestSuites === 0;
-        if (reporter.onRunComplete) {
-          reporter.onRunComplete(config, aggregatedResults);
-        }
-        return aggregatedResults;
+
+        return this._hasteMap.matchFiles(/\.snap/).then(res => {
```

Comment:
I think this code should be inside of jest-snapshot. Making jest-snapshot a dependency of jest-cli won't make a difference with npm3, so it's fine. I think the integration for the TestRunner should be something like `snapshots.cleanup();` but not more.